### PR TITLE
docs: Add maintainers guide

### PR DESCRIPTION
**Pull request by scottgerring**
**Issue #, if available:**
#1295 

## Description of changes:
Adds a maintainers guide to the documentation site, inspired by [python's maintainers playbook](https://docs.powertools.aws.dev/lambda/python/latest/maintainers/). 

**Checklist**

 

* [x] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda-java/#tenets)
* [ ] Update tests
* [x] Update docs
* [x] PR title follows [conventional commit semantics](https://www.conventionalcommits.org/en/v1.0.0/)

## Breaking change checklist

**RFC issue #**:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by sthulb — APPROVED** [review]
this is much improved

**Review by sthulb — APPROVED** [review]

File: docs/processes/maintainers.md [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+!!! note "Please treat this content as a living document."
+
+This is document explains who the maintainers are, their responsibilities, and how they should be doing it. If you're interested in contributing,
+ see [CONTRIBUTING](https://github.com/aws-powertools/powertools-lambda-java/blob/main/CONTRIBUTING.md){target="_blank"}.
+
+## Current Maintainers
+
+| Maintainer            | GitHub ID                                                                       | Affiliation |
+|-----------------------|---------------------------------------------------------------------------------| ----------- |
+| Jerome Van Der Linden | [sthulb](https://github.com/jeromevdl){target="_blank"}          | Amazon      |
```
Comment (jeromevdl) [inline docs/processes/maintainers.md:19]:
```suggestion
| Jerome Van Der Linden | [jeromevdl](https://github.com/jeromevdl){target="_blank"}          | Amazon      |
```

Comment (jeromevdl) [reply docs/processes/maintainers.md:19]: should we put AWS, rather than Amazon?

File: docs/processes/maintainers.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+| Scott Gerring         | [scottgerring](https://github.com/scottgerring){target="_blank"} | Amazon      |
+
+## Emeritus
+
+Previous active maintainers who contributed to this project.
+
+| Maintainer     | GitHub ID                                                                               | Affiliation   |
+|----------------|-----------------------------------------------------------------------------------------|---------------|
+| Mark Sailes    | [msailes](https://github.com/msailes){target="_blank"}                   | Amazon        |
+| Pankaj Agrawal | [ pankajagrawal16 ](https://github.com/pankajagrawal16){target="_blank"} | Former Amazon |
```
Comment (jeromevdl) [inline docs/processes/maintainers.md:30]:
```suggestion
| Pankaj Agrawal | [pankajagrawal16](https://github.com/pankajagrawal16){target="_blank"} | Former Amazon |
```

File: docs/processes/maintainers.md [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+### Triage RFCs
+
+RFC is a collaborative process to help us get to the most optimal solution given the context. Their purpose is to ensure
+everyone understands what this context is, their trade-offs, and alternative solutions that were part of the research 
+before implementation begins.
+
+Make sure you ask these questions in mind when reviewing:
+
+- Does it use our [RFC template](https://github.com/aws-powertools/powertools-lambda-java/issues/new?assignees=&labels=RFC%2C+triage&projects=&template=rfc.md&title=RFC%3A+)?
+- Does the match our [Tenets](https://docs.powertools.aws.dev/lambda/java/latest/#tenets)?
```
Comment (jeromevdl) [inline docs/processes/maintainers.md:152]:
```suggestion
- Does it match our [Tenets](https://docs.powertools.aws.dev/lambda/java/latest/#tenets)?
```

File: docs/processes/maintainers.md (hunk 1 above)
Comment (scottgerring) [reply docs/processes/maintainers.md:19]: [Python says Amazon](https://docs.powertools.aws.dev/lambda/python/latest/maintainers/), and I figure we're one company !

File: docs/processes/maintainers.md [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+Firstly, make sure the commit history in the `main` branch **(1)** it's up to date, **(2)** commit messages are semantic, 
+and **(3)** commit messages have their respective area, for example `feat: <change>`, `chore: ...`).
+
+**Looks good, what's next?**
+
+Kickoff the `Prepare for maven central release` workflow with the intended rekease version. Once this has completed, it will
+draft a Pull Request named something like `chore: Prep release 1.19.0`. the PR will **(1)** roll all of the POM versions
+forward to the new release version and **(2)** release notes. 
+
+Once this is done, check out the branch and clean up the release notes. These will be used both in the release notes
```
Comment (jeromevdl) [inline docs/processes/maintainers.md:178]: mention the changelog

File: docs/processes/maintainers.md [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+file and the [published github release information](https://github.com/aws-powertools/powertools-lambda-java/releases),
+and you can use the existing release notes to see how changes are summarized.
+
+Next, commit and push, wait for the build to complete, and merge to main. Once main has built successfully, create a
```
Comment (jeromevdl) [inline docs/processes/maintainers.md:182]:
```suggestion
Next, commit and push, wait for the build to complete, and merge to main. Once main has built successfully (i.e. end-to-end tests should pass), create a 
```

File: docs/processes/maintainers.md [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+Once this is done, check out the branch and clean up the release notes. These will be used both in the release notes
+file and the [published github release information](https://github.com/aws-powertools/powertools-lambda-java/releases),
+and you can use the existing release notes to see how changes are summarized.
+
+Next, commit and push, wait for the build to complete, and merge to main. Once main has built successfully, create a 
+tagged release from the Github UI, using the same release notes.
+
+Next, run the `Publish package to the Maven Central Repository` action to release the library.
+
+Finally, by hand, create a PR rolling all of the POMs onto the next snapshot version.
```
Comment (jeromevdl) [inline docs/processes/maintainers.md:187]:
```suggestion
Finally, by hand, create a PR rolling all of the POMs onto the next snapshot version (e.g. `1.20.0-SNAPSHOT`). 
```

**Review by jeromevdl — CHANGES_REQUESTED** [review]
few changes and it's ok

**Comment by scottgerring** [comment]
@jeromevdl should be good now!

File: docs/processes/maintainers.md [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+!!! note "Please treat this content as a living document."
+
+This is document explains who the maintainers are, their responsibilities, and how they should be doing it. If you're interested in contributing,
+ see [CONTRIBUTING](https://github.com/aws-powertools/powertools-lambda-java/blob/main/CONTRIBUTING.md){target="_blank"}.
+
+## Current Maintainers
+
+| Maintainer            | GitHub ID                                                                       | Affiliation |
+|-----------------------|---------------------------------------------------------------------------------| ----------- |
+| Jerome Van Der Linden | [sthulb](https://github.com/jeromevdl){target="_blank"}          | Amazon      |
```
Comment (jeromevdl) [inline docs/processes/maintainers.md:19]:
```suggestion
| Jerome Van Der Linden | [jeromevdl](https://github.com/jeromevdl){target="_blank"}          | Amazon      |
```

**Pull request merged**